elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -76,11 +77,45 @@ public ActivePermissionExecutor(boolean verbose, final RequestScope requestScope
         this.verbose = verbose;
     }
 
+    //The method returns the annotation that has to be used to evaluate the specific action for the resource
+    //
+    private Class<? extends Annotation> getSecurityAnnotationType(EntityAction action) {
+        switch (action) {
+            case CREATE:
+                return CreatePermission.class;
+            case READ:
+                return ReadPermission.class;
+            case SHARE:
```

Comment:
We already have a PR merged for falling back share permission. Why are we implementing this again here?

Reply:
We were using the annotation class objects to represent the action that we intend to perform on the object. We have seen some cases recently where the annotation expression that we try to evaluate is different than the annotation that we specify as the argument. e.g. We evaluate @ReadPermission whenever we pass @SharePermission and We evaluate @CreatePermission when we call @UpdatePermission on newly created entities. 
So I felt like having a different set of enums for representing the actions would signify the difference between actions on entities and the security expressions that we evaluate.
When I refactored the the code around this idea, I had the method to map UPDATE action to @CreatePermission for newly created entities. So the action of mapping SHARE action to @ReadPermission seemed similar, so I moved the logic to the same method.